app/views/admin/assets/show.html.erb

```diff
@@ -24,7 +24,7 @@
 
     <% if @asset.file %>
       <%= link_to "Download Original",
-            @asset.file.url(response_content_disposition: ContentDisposition.format(disposition: :attachment, filename: @asset.title)),
+            download_path(@asset),
```

Comment:
Hmm, this changes things to through the downloads controller now. So the URL is pointing to an app URL, which will then redirect you to the S3 location. 

While convenient, I was thinking that the admin download link ought to be _directly_ to s3 like it was before. To avoid any bugs or problems that might be in the downloads controller, and to be a little bit more 'raw' access for admins. 

Also because while presently we want the same content-disposition filename, that's sort of just an "accident", I don't think they should actually be tied together, it should be easy to change our mind about admin download content-disposition filename separate from public. 

So, I'm thinking the same code that was there before -- don't switch to pointing to our local downloads controller -- just where before it said `filename: @asset.title`, switch to `filename: x`, where `x` is whatever logic the downloads controller uses to figure out filename. Just re-use that filename-determining logic, instead of actually pointing to an in-app URL. 

Does that seem reasonable?

Reply:
Absolutely reasonable. I hesitated about it, but I see your point Will do.

Reply:
Fixed.